sunxi: add testing 5.4 kernel

abajk opened the pull request:
These patches was removed, because they are already in mainline 5.4 kernel:
* 010-v5.3-drivers-ata-ahci_sunxi-Increased-SATA-AHCI-DMA-TX-RX.patch
* 101-arm64-dts-allwinner-a64-Enable-A64-timer-workaround.patch

Compiled and tested on Orange PI One (cortexa7) and Orange Pi PC2 (cortexa53).

abajk committed:
sunxi: copy config and patches from 4.19 to 5.4

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

ynezz commented on target/linux/generic/config-5.4:
```diff
@@ -170,10 +170,12 @@ CONFIG_ANON_INODES=y
 # CONFIG_AR8216_PHY is not set
```
Changes to the generic config should go into separate commit.

ynezz commented on target/linux/sunxi/config-5.4:
```diff
@@ -1,24 +1,32 @@
+CONFIG_64BIT_TIME=y
 # CONFIG_AHCI_SUNXI is not set
 CONFIG_ALIGNMENT_TRAP=y
-# CONFIG_ARCH_AXXIA is not set
```
It would be nice to get the refresh of configs into separate commits and due to the number of changes in this file (+48 -53) and a53's I would probably try to make the review easier by splitting it into separate commits per subtarget as well.

ynezz commented on target/linux/sunxi/patches-5.4/010-v5.3-drivers-ata-ahci_sunxi-Increased-SATA-AHCI-DMA-TX-RX.patch:
```diff
@@ -1,94 +0,0 @@
-From 120357ea176e420d313cf8cf2ff35fbe233d3bab Mon Sep 17 00:00:00 2001
```
Refresh of patches should go into separate commit as well.

ynezz commented on target/linux/sunxi/config-5.4:
```diff
@@ -0,0 +1,605 @@
+# CONFIG_AHCI_SUNXI is not set
```
Nitpick, it's obvious, but you're saving the commiter some time if you add missing commit description, otherwise the commiter is going to add one anyway.

olek210 commented:
[WiP]
Commit "kernel: sort generic configuration" moved into separate [PR](https://github.com/openwrt/openwrt/pull/2817).

ynezz commented:
> Commit "kernel: sort generic configuration" moved into separate PR.

Uh, why? Did I wrote separate PR? I wrote separate *commit* :-)

abajk committed:
sunxi: make patches apply cleanly on 5.4

These patches are already in mainline 5.4 kernel:
* 010-v5.3-drivers-ata-ahci_sunxi-Increased-SATA-AHCI-DMA-TX-RX.patch
* 101-arm64-dts-allwinner-a64-Enable-A64-timer-workaround.patch

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
kernel: move disabled symbols into generic kconfig

Move some disabled kconfig options found in sunxi kconfig into generic.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: add testing 5.4 kernel

Add 5.4 as testing kernel.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: remove options that are already in generic kconfig

These options are already definied in generic 5.4 kconfig.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: remove old symbols from 5.4 kconfig

These options only exist earlier kernel versions.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: refresh config for all subtargets

This was done by executing these command for all subtargets:

$ time make -j9 kernel_oldconfig CONFIG_TARGET=subtarget

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

abajk committed:
sunxi: move common options from subtargets into target kconfig

These new options are common for all subtargets and can be moved into target kconfig.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

olek210 commented:
@ynezz Done. Patches are splitted into smaller pieces. Now should be easier to review patches :)
I now, but it can be merged independently and generic config changes very often.

ynezz commented:
Merged, thanks!
